ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -24,12 +24,65 @@ void dumpValueAndExit(const char *Msg, Value *V) {
   exit(EXIT_FAILURE);
 }
 
-std::vector<Value *> getTraceInputs(Function *F, uintptr_t BBIdx) {
+// A function name and basic block index pair that identifies a block in the
+// AOT LLVM IR.
+struct IRBlock {
+  // A non-null pointer to the function name.
+  char *FuncName;
+  // The index of the block in the parent LLVM function.
+  size_t BBIdx;
+};
+
+// Describes the software or hardware trace to be compiled using LLVM.
```

Comment:
"Describes" seems an odd term here: I'm actually not sure that the struct is accurately named? Judging by the class fields, it seems more like a "block to function" mapper or ... ?

Reply:
It really does express the collected trace. The pair `(FuncNames[Idx], BBs[Idx])` identifies the LLVM IR block at position `Idx` in the trace (as long as `FuncNames[Idx] != nullptr`).

 The choice of data structures is a little odd due to shoe-horning through the C ABI (whilst trying not to add a new struct type to both the C side and the rust side).

Reply:
We could add a `repr(C)` type to Rust and a mirror type to C++ if you think that's cleaner?

Reply:
> We could add a repr(C) type to Rust and a mirror type to C++ if you think that's cleaner?

I have no idea! Please do as you see fit here.

Reply:
What do you think @ptersilie

Reply:
There is perhaps a small performance advantage to using a repr(C) type, as we then don't have to materialise them from an array once in C++...

Reply:
Maybe that's something for a future PR?

Reply:
If you think it's a future PR thing, then let's file an issue for it and move on?

Reply:
I guess https://github.com/ykjit/yk/issues/419 is the relevant issue?

Reply:
Yes. I added a comment with a link in bfa1c25.